cases/fee.optional.test.js

```diff
@@ -54,18 +54,37 @@ describe("Fee", () => {
     expect(response.headers.get("access-control-allow-origin")).toBe("*");
   });
 
+  it("returns error for request with no authorization header if fee_required", async () => {
+    if (needFeeAuth) {
+      let response = await fetch(toml.TRANSFER_SERVER + "/fee");
+      expect(response.status).toBeGreaterThanOrEqual(400);
+      expect(response.status).toBeLessThan(500);
+      expect(await response.json()).toMatchSchema(errorSchema);
+    }
+  });
+
   it("returns a proper error schema for a non-params fee request", async () => {
-    const response = await fetch(toml.TRANSFER_SERVER + "/fee");
+    const headers = needFeeAuth
+      ? { Authorization: `Bearer ${jwt}` }
```

Comment:
do we not need origin for authorized calls?

Reply:
I don't believe so but I'll double check.

Reply:
testing `testanchor.stellar.org`, all `/fee` tests are successful and it requires auth so I think this is the case. It does seem a bit weird though, let me see if I can find an explanation.

Reply:
https://stackoverflow.com/questions/10636611/how-does-access-control-allow-origin-header-work

If the request is "non-simple", as the answer puts it, there are exceptions to the typical CORS logic. One way a request can be non-simple is if it has a header that isn't in this list:
- Accept
- Accept-Language
- Content-Language
- Content-Type (this is only simple when its value is application/x-www-form-urlencoded, multipart/form-data, or text/plain)

So since `Authorization` isn't a simple header the `Origin` header isn't required. This actually makes sense to me. If a request has the necessary authentication token it shouldn't be blocked by CORS, since the server has essentially already given permission to the client.